src/Lecture3.hs

```diff
@@ -181,7 +213,14 @@ together only different elements.
 Product {getProduct = 6}
 
 -}
-appendDiff3 = error "TODO"
+
+appendDiff3 ::(Eq a, Semigroup a) => a -> a -> a -> a
+appendDiff3 a1 a2 a3
+  | a1 == a2 && a2 == a3 = a1
+  | a1 == a2 && a2 /= a3 = a1 <> a3
+  | a1 /= a2 && a2 == a3 = a1 <> a2
+  | a1 /= a2 && a1 == a3 = a1 <> a2
```

Comment:
Since guards are checked top-down (from the first one to the last one), you can simplify some checks as you already know that some conditions are already false 🎌 
```suggestion
  | a1 == a2 = a1 <> a3
  | a2 == a3 = a1 <> a2
  | a1 == a3 = a1 <> a2
```